pkg/sanity/osd_test_util.go

```diff
@@ -65,11 +76,29 @@ func testIfVolumeCreatedSuccessfully(
 	Expect(volumesList[0].GetSpec().GetJournal()).To(BeEquivalentTo(vr.GetSpec().GetJournal()))
 	Expect(volumesList[0].GetSpec().GetNfs()).To(BeEquivalentTo(vr.GetSpec().GetNfs()))
 	Expect(volumesList[0].GetSpec().GetPassphrase()).To(BeEquivalentTo(vr.GetSpec().GetPassphrase()))
-	//Expect(volumesList[0].GetSpec().GetReplicaSet()).To(BeEquivalentTo(vr.GetSpec().GetNfs()))
+	Expect(volumesList[0].GetSpec().GetReplicaSet()).To(BeEquivalentTo(vr.GetSpec().GetReplicaSet()))
 	Expect(volumesList[0].GetSpec().GetScale()).To(BeEquivalentTo(vr.GetSpec().GetScale()))
 	Expect(volumesList[0].GetSpec().GetShared()).To(BeEquivalentTo(vr.GetSpec().GetShared()))
-	//Expect(volumesList[0].GetSpec().GetSize()).To(BeEquivalentTo(vr.GetSpec().GetSize()))
+	Expect(volumesList[0].GetSpec().GetSize()).To(BeEquivalentTo(vr.GetSpec().GetSize()))
 	Expect(volumesList[0].GetSpec().GetSnapshotInterval()).To(BeEquivalentTo(vr.GetSpec().GetSnapshotInterval()))
 	Expect(volumesList[0].GetSpec().GetSnapshotSchedule()).To(BeEquivalentTo(vr.GetSpec().GetSnapshotSchedule()))
 	Expect(volumesList[0].GetSpec().GetSticky()).To(BeEquivalentTo(vr.GetSpec().GetSticky()))
 }
+
+//Returns an in between min and max. Min - included, Max excluded. So mathematically [min, max)
+func random(min, max int) int {
+	if max == min {
+		return max
+	}
+	rand.Seed(time.Now().Unix())
+	return rand.Intn(max-min) + min
+}
+
+func getNewHALevel(currentHA int64) (newHALevel int) {
+	if currentHA == 1 || currentHA == 3 {
+		newHALevel = 2
+	} else {
+		newHALevel = []int{1, 3}[random(0, 2)]
```

Comment:
I don't understand this. Why only randomize when HA is 2?
You could always randomize within a set of numbers which does not contain the current HA level.

Reply:
As discussed ,  I will remove the random HA- update and make it simple increment of 1 for this PR.
Next PR will address this issue and further tests of having more combinations of creating with different replication factor combinations and replification factor updates.